ios/brave-ios/Sources/Brave/Frontend/Browser/New Tab Page/NewTabPageViewController.swift

```diff
@@ -257,7 +257,18 @@ class NewTabPageViewController: UIViewController {
     }
 
     background.changed = { [weak self] in
-      self?.setupBackgroundImage()
+      guard let self else { return }
+      setupBackgroundImage()
+
+      let isTabVisible = viewIfLoaded?.window != nil
+      setupBackgroundVideoIfNeeded(shouldCreatePlayer: isTabVisible)
```

Comment:
Here is a proper place to call `setupBackgroundVideoIfNeeded` after background is changed (which happens when user changes `New Tab Page` setting)

Reply:
We also need to create AVPlayer object only if current tab is visible to avoid memory/processor consumption.
AVPlayer object will be created during `willMove(...)` when user switches to tab.